### PR TITLE
bpo-42115: Add an opcode cache for LOAD_METHOD

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue42115

### CODE REVIEW

File: Python/ceval.c [hunk 1]
```diff
@@ -111,7 +111,7 @@ static long dxp[256];
 #else
 #define OPCACHE_MIN_RUNS 1024  /* create opcache when code executed this time */
 #endif
-#define OPCODE_CACHE_MAX_TRIES 20
+#define OPCACHE_MAX_TRIES 20
```
Comment (1st1) [inline Python/ceval.c:114]: Let's maybe add a comment clarifying what this knob does?

**Review by 1st1 — APPROVED** [review]

**Comment by 1st1** [comment]
Add a NEWS update?

Also this looks pretty similar to my original patch but I didn't compare. Were there any changes to it, I'm just curious?

**Comment by pablogsal** [comment]
> Also this looks pretty similar to my original patch but I didn't compare. Were there any changes to it, I'm just curious?

The idea is the same but this part is simplified:

https://github.com/python/cpython/pull/23503/files#diff-c22186367cbe20233e843261998dc027ae5f1f8c0d2e778abfa454ae74cc59deR3866-R3875

I am also using `_PyDict_GetItem_KnownHash` to fetch the items for slightly better performance and the rest is just modifications to account for the code changes around.

I am waiting for Inada-san to confirm the benchmarks before landing

**Comment by pablogsal** [comment]
I am closing this as the benchmarks are not that exiting after rebasing and benchmarking again :(

**Pull request closed without merging**